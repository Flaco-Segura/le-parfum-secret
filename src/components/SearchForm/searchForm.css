.search-form {
  padding: 10px 0 20px 0;
  display: flex;
  align-items: baseline;
}

.search-form > p {
  margin-left: 30px;
}