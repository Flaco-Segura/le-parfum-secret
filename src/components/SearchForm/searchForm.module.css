.searchForm {
  padding: 10px 0 20px 0;
  display: flex;
  align-items: baseline;
}

.searchForm > p {
  margin-left: 30px;
}

.searchFormLine {
  display: flex;
  justify-content: flex-start;
}