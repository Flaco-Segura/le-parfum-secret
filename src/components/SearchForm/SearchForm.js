import * as React from 'react';
import { InputWithLabel } from '../InputWithLabel/InputWithLabel';

import './searchForm.css';

const SearchForm = ({searchTerm, handleSearchInput, handleSearchSubmit}) => {
  return <form onSubmit={handleSearchSubmit} className='search-form'>
    <InputWithLabel
      htmlFor='search'
      id='search'
      value={searchTerm}
      isFocused
      onInputChange={handleSearchInput}
    >
      <strong>Search: </strong>
    </InputWithLabel>
    <button
      type='submit'
      disabled={!searchTerm}
      className='button button_large'  
    >
      Submit
    </button>
    <p>
      Searching for <strong>{searchTerm}</strong>
    </p>
  </form>
}

export { SearchForm };
