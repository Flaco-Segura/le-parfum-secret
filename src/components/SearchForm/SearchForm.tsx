import * as React from 'react';
import InputWithLabel from '../InputWithLabel/InputWithLabel';

import './searchForm.css';

import { SearchFormProps } from '../../types/SearchFormType';

const SearchForm = ({searchTerm, onSearchInput, onSearchSubmit}: SearchFormProps) => {
  return <form onSubmit={onSearchSubmit} className='search-form'>
    <InputWithLabel
      // htmlFor='search'
      id='search'
      value={searchTerm}
      isFocused
      onInputChange={onSearchInput}
    >
      <strong>Search: </strong>
    </InputWithLabel>
    <button
      type='submit'
      disabled={!searchTerm}
      className='button button_large'  
    >
      Submit
    </button>
    <p>
      Searching for <strong>{searchTerm}</strong>
    </p>
  </form>
}

export default SearchForm;
