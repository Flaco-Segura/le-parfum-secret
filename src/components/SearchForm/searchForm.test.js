import * as React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';

import SearchForm from './SearchForm';

describe('SearchForm', () => {
  const searchFormProps = {
    searchTerm: 'React',
    handleSearchInput: jest.fn(),
    handleSearchSubmit: jest.fn(),
  };

  test('renders the input field with its value', () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByDisplayValue('React')).toBeInTheDocument();
  });

  test('renders the correct label', () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByLabelText(/Search/)).toBeInTheDocument();
  });

  test('calls onSearchInput on input field change', () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.change(screen.getByDisplayValue('React'), {
      target: { value: 'Redux' },
    });

    expect(searchFormProps.handleSearchInput).toHaveBeenCalledTimes(1);
  });

  test('calls onSearchSubmit on button submit click', () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.submit(screen.getByRole('button'));

    expect(searchFormProps.handleSearchSubmit).toHaveBeenCalledTimes(1);
  });
});
