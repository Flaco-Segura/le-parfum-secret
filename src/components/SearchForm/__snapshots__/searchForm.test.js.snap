// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders snapshot 1`] = `
<form
  class="search-form"
>
  <label
    class="label"
    for="search"
  >
    <strong>
      Search: 
    </strong>
  </label>
  Â 
  <div
    class="inputwithlabel_container"
  >
    <input
      class="continer__input"
      id="search"
      type="text"
      value="React"
    />
    <div
      class="search_icon_container"
    >
      <div
        class="search_icon"
      />
    </div>
  </div>
  <button
    class="button button_large"
    type="submit"
  >
    Submit
  </button>
  <p>
    Searching for 
    <strong>
      React
    </strong>
  </p>
</form>
`;
